In [10]:
#Importing Required Libraries
import numpy as np
import pandas as pd

In [11]:
#Reading Dataset
df=pd.read_csv("news.csv")

In [12]:
#Accessing First 5 Elements of Dataset
df.head()

Unnamed: 0                                              title  \
0        8476                       You Can Smell Hillary’s Fear   
1       10294  Watch The Exact Moment Paul Ryan Committed Pol...   
2        3608        Kerry to go to Paris in gesture of sympathy   
3       10142  Bernie supporters on Twitter erupt in anger ag...   
4         875   The Battle of New York: Why This Primary Matters   

                                                text label  
0  Daniel Greenfield, a Shillman Journalism Fello...  FAKE  
1  Google Pinterest Digg Linkedin Reddit Stumbleu...  FAKE  
2  U.S. Secretary of State John F. Kerry said Mon...  REAL  
3  — Kaydee King (@KaydeeKing) November 9, 2016 T...  FAKE  
4  It's primary day in New York and front-runners...  REAL

In [13]:
#Checking for Null Values
df.isnull().sum()

Unnamed: 0    0
title         0
text          0
label         0
dtype: int64

In [14]:
#Extracting Independent and Dependent Variables
x=df.iloc[:,2]
y=df.iloc[:,3]

In [15]:
#Splitting Dataset into Train and Test Set
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2,random_state=7)

In [16]:
#Fitting TfidfVectorizer through Train and Test Set
from sklearn.feature_extraction.text import TfidfVectorizer
tfidf=TfidfVectorizer(stop_words="english",max_df=0.7)

x_train=tfidf.fit_transform(x_train)
x_test=tfidf.transform(x_test)

In [22]:
#Fitting PassiveAggressiveClassifier through Train set
from sklearn.linear_model import PassiveAggressiveClassifier
clf=PassiveAggressiveClassifier(max_iter=180)
clf.fit(x_train,y_train)

PassiveAggressiveClassifier(max_iter=180)

In [23]:
#Predicting Test Set
y_pred=clf.predict(x_test)

In [24]:
#Calculating Accuracy Score
from sklearn.metrics import accuracy_score
acscore=accuracy_score(y_test,y_pred)
print(acscore)

0.9273875295974744


Thus the accuracy score of our model is 93%.

In [25]:
#Creating the Confusion Matrix
from sklearn.metrics import confusion_matrix
cm=confusion_matrix(y_test,y_pred)
print(cm)

[[588  50]
 [ 42 587]]


Thus our model gives us 588+587=1175 correct predictions and 50+42=92 incorrect predictions.

In [26]:
#Creating the Classification Report
from sklearn.metrics import classification_report
cr=classification_report(y_test,y_pred)
print(cr)

              precision    recall  f1-score   support

        FAKE       0.93      0.92      0.93       638
        REAL       0.92      0.93      0.93       629

    accuracy                           0.93      1267
   macro avg       0.93      0.93      0.93      1267
weighted avg       0.93      0.93      0.93      1267



Thus, the average Precision, Recall and F1 score of our model is 0.93